WordPress/gutenberg
# [Widgets editor] Disallow multiple instances of reference widgets

## Description

This PR fixes #25494 by:

* Showing an error message if such reference widget already exists in another widget area
* Removing duplicates from submitted API requests

This PR only affects the frontend and does **not** change the actual API handlers.

## How has this been tested?

1. Add the following code to gutenberg.php: https://gist.github.com/adamziel/526436b4cc9bc82a7221c77e00137ae1
1. Go to widgets editor
1. Add a marquee widget
1. Add another marquee widget
1. Confirm you see the same message as on the screenshots below
1. Add another legacy widget, could be by just copying&pasting the marquee and renaming it to e.g. "Calendar"
1. Repeat steps 3-5, confirm the error allows you to switch to a Calendar widget
1. Click save while the duplicates are still in the editor, confirm only one instance of each reference widget got saved

## Screenshots 

<img width="682" alt="Zrzut ekranu 2020-10-15 o 13 02 03" src="https://user-images.githubusercontent.com/205419/96115030-a943ec80-0ee6-11eb-90b3-e93f1812a4fb.png">
<img width="702" alt="Zrzut ekranu 2020-10-15 o 13 02 14" src="https://user-images.githubusercontent.com/205419/96115032-ab0db000-0ee6-11eb-9647-bc7ed4955513.png">

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
